sale_blanket_order/models/blanket_orders.py

```diff
@@ -298,11 +313,15 @@ def set_to_draft(self):
     def action_confirm(self):
         self._validate()
         for order in self:
-            sequence_obj = self.env["ir.sequence"]
-            if order.company_id:
-                sequence_obj = sequence_obj.with_company(order.company_id.id)
-            name = sequence_obj.next_by_code("sale.blanket.order")
-            order.write({"confirmed": True, "name": name})
+            update_data = {"confirmed": True}
+            # if the order already had a sequence number we don't need to set it again
+            if not self.env.company.blanket_order_seq_number_from_draft:
```

Comment:
**issue**: I believe the company here should not be `self.env.company` but `order.company_id` (and maybe `self.env.company` as a fallback, in case `order.company_id` is empty.

An alternative approach (that also sidesteps the `company` problem) could be to not check the setting at all, but instead replicate the check in the create: simply, if the field `name` is still empty/or it still says "Draft", give it a number. We for sure want it to have a number, we only need to make sure we're not overwriting an existing number.